jax/_src/prng.py

```diff
@@ -593,9 +591,17 @@ def device_put_replicated(val, aval, sharding, devices):
     physical_result = pxla.batched_device_put(physical_aval, physical_sharding, [physical_buf] * len(devices), devices)
     return random_wrap(physical_result, impl=aval.dtype._impl)
 
+  @staticmethod
   def tangent_dtype(_):
     return dtypes.float0
 
+  # TODO(mattjj,frostig): even though the key dtype shouldn't appear in
+  # tangents, our ad.replace_float0s in custom_jvp/vjp means passing in zeros
+  # like the primal to user rules
+  @staticmethod
+  def zero(aval):
+    return lax_internal.zeros_like_shaped_array(aval.update(dtype=dtypes.float0))
```

Comment:
Is it better to pass into `custom_jvp`/`vjp` a float0-dtyped array, a `SymbolicZero`, or some float0-like extended dtype that we could cook up?

Reply:
We can merge and think on it.